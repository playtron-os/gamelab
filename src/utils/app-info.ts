import {
  AppDownloadStage,
  AppInformation,
  AppStatus,
  AppProvider
} from "@/types";

import { PlaytronImage, InstalledApp } from "@/types/app-library";
import { t } from "@lingui/macro";
import { CloudFill, RunFill } from "@playtron/styleguide";

export function getProviderName(provider: string): string {
  return (
    {
      [AppProvider.Steam]: "Steam",
      [AppProvider.Gog]: "GOG",
      [AppProvider.EpicGames]: "Epic Games"
    }[provider] || provider[0].toUpperCase() + provider.substring(1)
  );
}

export function nearestPowerOfTwo(n: number) {
  const log = Math.round(Math.log(n) / Math.log(2));
  const a = Math.round(Math.pow(2, log));
  const b = Math.round(Math.pow(2, log + 1));

  return n - a < b - n ? a : b;
}

export function getDiskSize(size: number | undefined): string {
  if (!size) {
    return "";
  }
  const units = ["B", "KB", "MB", "GB"];
  let unitIndex = 0;
  while (size > 1024 && unitIndex < units.length - 1) {
    size /= 1024;
    ++unitIndex;
  }
  return `${Math.round(size * 10) / 10}${units[unitIndex]}`;
}

/**
 * Variant of getDiskSize to return size printed
 * on the disk's label such as 2TB or 256GB.
 */
export function getDiskSizeLabel(size: number): string {
  if (size === 0) {
    return "";
  }
  const units = ["B", "KB", "MB", "GB", "TB"];
  let unitIndex = 0;
  while (size > 990 && unitIndex < units.length - 1) {
    size /= 1000;
    ++unitIndex;
  }
  const diskSize = Math.round(Math.round(size * 100) / 100);
  const nearestSize = nearestPowerOfTwo(diskSize);
  const sizeRatio = diskSize / nearestSize;
  if (sizeRatio >= 0.99 || sizeRatio <= 1.01) {
    return `${nearestSize}${units[unitIndex]}`;
  }
  return `${diskSize}${units[unitIndex]}`;
}

export const getAppStatusLabel = (status: AppStatus) => {
  switch (status) {
    case AppStatus.READY:
      return t`Ready`;
    case AppStatus.PRE_ALLOCATING:
      return t`Pre-Allocating`;
    case AppStatus.VERIFIYING:
      return t`Verifying`;
    case AppStatus.DOWNLOADING:
      return t`Downloading`;
    case AppStatus.INSTALLING:
      return t`Installing`;
    case AppStatus.UPDATE_REQUIRED:
      return t`Update Required`;
    case AppStatus.PAUSED:
      return t`Paused`;
    case AppStatus.QUEUED:
      return t`Queued`;
    case AppStatus.NOT_DOWNLOADED:
      return t`Not Downloaded`;
    case AppStatus.RUNNING:
      return t`Running`;
    case AppStatus.LAUNCHING:
      return t`Launching`;
    default:
      return t`Unknown`;
  }
};

export const getProgress = (installedApp?: InstalledApp): number => {
  if (!installedApp) return 0;
  switch (installedApp.download_status.stage) {
    case AppDownloadStage.DOWNLOADING:
    case AppDownloadStage.PRE_ALLOCATING:
    case AppDownloadStage.VERIFYING:
      return installedApp.download_status.progress || 0;
    case AppDownloadStage.INSTALLING:
      return (
        (installedApp.post_install.reduce(
          (prev, val) => prev + (val.is_success ? 1 : 0),
          0
        ) /
          installedApp.post_install.length) *
        100
      );
    default:
      return 0;
  }
};

export function getImage(images: PlaytronImage[]): string {
  if (!images) {
    return "";
  }
  let gogCandidate = "";
  let steamCandidate = "";
  let epicGamesCandidate = "";
  for (const image of images) {
    if (image.source === "steam" && image.image_type === "header") {
      steamCandidate = image.url;
    }

    if (
      image.source === "gog" &&
      image.url != "https" &&
      image.image_type === "logo"
    ) {
      gogCandidate = image.url.replace("https//", "https://");
    }
    if (image.source === "epicgames" && image.image_type === "OfferImageWide") {
      epicGamesCandidate = image.url;
    }
  }
  if (gogCandidate) {
    return gogCandidate;
  }
  if (epicGamesCandidate) {
    return epicGamesCandidate;
  }
  if (steamCandidate) {
    return steamCandidate;
  }

  return images.length.toString();
}

export function getDate(date: Date | number): string {
  return new Date(date).toDateString();
}

export const getAppActionLabelByStatus = (status: AppStatus) => {
  switch (status) {
    case AppStatus.READY:
      return t`Launch`;
    case AppStatus.PRE_ALLOCATING:
    case AppStatus.VERIFIYING:
    case AppStatus.DOWNLOADING:
      return t`Pause`;
    case AppStatus.NOT_DOWNLOADED:
      return t`Download`;
    case AppStatus.UPDATE_REQUIRED:
      return t`Update`;
    case AppStatus.PAUSED:
      return t`Resume`;
    case AppStatus.QUEUED:
      return t`Prioritize`;
    case AppStatus.RUNNING:
      return t`Stop`;
    case AppStatus.LAUNCHING:
      return t`Launching...`;
    default:
      return "...";
  }
};

/**
 * Get the App status without the QUEUED status which is not accessible from the AppInformation object
 * To get the AppStatus including the QUEUED status use getAppStatusWithQueue.
 *
 * @param appInfo
 * @returns AppStatus
 */
export const getAppStatus = (appInfo: AppInformation): AppStatus => {
  if (appInfo.is_running) {
    return AppStatus.RUNNING;
  } else if (appInfo.is_launched) {
    return AppStatus.LAUNCHING;
  }

  if (appInfo.is_installing) {
    return AppStatus.INSTALLING;
  }

  if (appInfo.is_downloading) {
    switch (appInfo.installed_app?.download_status.stage) {
      case AppDownloadStage.PRE_ALLOCATING:
        return AppStatus.PRE_ALLOCATING;
      case AppDownloadStage.VERIFYING:
        return AppStatus.VERIFIYING;
      default:
        return AppStatus.DOWNLOADING;
    }
  }

  switch (appInfo.installed_app?.download_status.stage) {
    case AppDownloadStage.PRE_ALLOCATING:
    case AppDownloadStage.VERIFYING:
    case AppDownloadStage.DOWNLOADING:
      return AppStatus.PAUSED;
    case AppDownloadStage.UPDATE_REQUIRED:
      return AppStatus.UPDATE_REQUIRED;
    case AppDownloadStage.DONE:
      return AppStatus.READY;
    default:
      return AppStatus.NOT_DOWNLOADED;
  }
};

/**
 * Return AppStatus, including QUEUED. Make sure to provide the necessary React context.
 * @param appInfo
 * @param queuePositionMapState
 * @returns AppStatus
 */
export const getAppStatusWithQueue = (
  appInfo: AppInformation,
  queuePositionMapState: { [key: string]: number }
): AppStatus => {
  const status = getAppStatus(appInfo);
  if (
    [
      AppStatus.RUNNING,
      AppStatus.INSTALLING,
      AppStatus.DOWNLOADING,
      AppStatus.PRE_ALLOCATING,
      AppStatus.VERIFIYING
    ].includes(status)
  ) {
    return status;
  }
  if (
    appInfo.installed_app &&
    typeof queuePositionMapState[appInfo.installed_app.owned_app.id] ===
      "number"
  ) {
    return AppStatus.QUEUED;
  }
  return status;
};

export const getAppActionIconByStatus = (status: AppStatus) => {
  if (status === AppStatus.READY) {
    return RunFill;
  }
  if (status === AppStatus.UPDATE_REQUIRED) {
    return CloudFill;
  }
};

/**
 * Return a user friendly label for a drive.
 * Uses the last part of the mount point or return "System Drive" for the root mount point.
 * @param drive
 * @returns string
 */
export const getDriveLabel = (drive: string | undefined): string => {
  if (!drive) {
    return "";
  }
  if (drive === "/") {
    return t`System Drive`;
  }

  const label = drive.split("/").pop() || drive;
  return label.replace(".ext4", "");
};
