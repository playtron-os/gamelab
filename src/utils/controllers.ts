import { ControllerInfo, InputEvent } from "@/types/input-config";
import { t } from "@lingui/macro";
import { physicalLayouts } from "@/constants/physical-layouts";

export const KEYBOARD_QWERTY_LAYOUT = [
  [
    "KeyEsc",
    "KeyF1",
    "KeyF2",
    "KeyF3",
    "KeyF4",
    "KeyF5",
    "KeyF6",
    "KeyF7",
    "KeyF8",
    "KeyF9",
    "KeyF10",
    "KeyF11",
    "KeyF12"
  ],
  [
    "KeyGrave",
    "Key1",
    "Key2",
    "Key3",
    "Key4",
    "Key5",
    "Key6",
    "Key7",
    "Key8",
    "Key9",
    "Key0",
    "KeyMinus",
    "KeyEqual",
    "KeyBackspace"
  ],
  [
    "KeyTab",
    "KeyQ",
    "KeyW",
    "KeyE",
    "KeyR",
    "KeyT",
    "KeyY",
    "KeyU",
    "KeyI",
    "KeyO",
    "KeyP",
    "KeyLeftBrace",
    "KeyRightBrace",
    "KeyBackslash"
  ],
  [
    "KeyCapslock",
    "KeyA",
    "KeyS",
    "KeyD",
    "KeyF",
    "KeyG",
    "KeyH",
    "KeyJ",
    "KeyK",
    "KeyL",
    "KeySemicolon",
    "KeyApostrophe",
    "KeyEnter"
  ],
  [
    "KeyLeftShift",
    "KeyZ",
    "KeyX",
    "KeyC",
    "KeyV",
    "KeyB",
    "KeyN",
    "KeyM",
    "KeyComma",
    "KeyDot",
    "KeySlash",
    "KeyRightShift"
  ],
  [
    "KeyLeftCtrl",
    "KeyLeftMeta",
    "KeyLeftAlt",
    "KeySpace",
    "KeyRightAlt",
    "KeyCompose",
    "KeyRightMeta",
    "KeyRightCtrl"
  ]
];

export const KEYBOARD_NUMPAD_LAYOUT = [
  [
    "KeyInsert",
    "KeyHome",
    "KeyPageUp",
    "KeyPlay",
    "KeyStop",
    "KeyNext",
    "KeyPause"
  ],
  [
    "KeyDelete",
    "KeyEnd",
    "KeyPageDown",
    "KeyEqual",
    "KeySlash",
    "KeyAsterisk",
    "KeyMinus"
  ],
  [null, null, null, "KeyKp7", "KeyKp8", "KeyKp9", "KeyPlus"],
  [null, null, null, "KeyKp4", "KeyKp5", "KeyKp6"],
  [null, "KeyUp", null, "KeyKp1", "KeyKp2", "KeyKp3", "KeyEnter"],
  ["KeyLeft", "KeyDown", "KeyRight", "KeyKp0", "KeyDot"]
];

export const getKeyWidth = (key: string) => {
  switch (key) {
    case "KeyGrave":
      return "w-[68px]";
    case "KeyBackspace":
      return "w-[76px]";
    case "KeyTab":
      return "w-[96px]";
    case "KeyEsc":
    case "KeyF1":
    case "KeyF2":
    case "KeyF3":
    case "KeyF4":
    case "KeyF5":
    case "KeyF6":
    case "KeyF7":
    case "KeyF8":
    case "KeyF9":
    case "KeyF10":
    case "KeyF11":
    case "KeyF12":
      return "w-[55.71px]";
    case "KeyCapslock":
      return "w-[111px]";
    case "KeyRightShift":
    case "KeyLeftShift":
      return "w-[124px]";
    case "KeyEnter":
      return "w-[85px]";
    case "KeyRightCtrl":
    case "KeyLeftCtrl":
    case "KeyRightMeta":
    case "KeyLeftMeta":
    case "KeyCompose":
    case "KeyRightAlt":
    case "KeyLeftAlt":
      return "w-[65px]";
    case "KeySpace":
      return "w-[289px]";
    default:
      return "w-12";
  }
};

export const getKeyLabel = (key: string | null) => {
  if (!key) {
    return t`Unset`;
  }
  switch (key) {
    case "KeyGrave":
      return "`";
    case "KeySpace":
      return "Space";
    case "KeyBackspace":
      return "⇐";
    case "KeyEnter":
      return "⏎";
    case "KeyLeftShift":
    case "KeyRightShift":
      return "Shift";
    case "KeyLeftMeta":
    case "KeyRightMeta":
      return "Win";
    case "KeyLeftCtrl":
    case "KeyRightCtrl":
      return "Ctrl";
    case "KeyRightAlt":
    case "KeyLeftAlt":
      return "Alt";
    case "KeyLeftBrace":
      return "[";
    case "KeyRightBrace":
      return "]";
    case "KeyEqual":
      return "=";
    case "KeyComma":
      return ",";
    case "KeyDot":
      return ".";
    case "KeySlash":
      return "/";
    case "KeyBackslash":
      return "\\";
    case "KeyCompose":
      return "Fn";
    case "KeyApostrophe":
      return "'";
    case "KeySemicolon":
      return ";";
    case "KeyMinus":
      return "-";
    case "KeyInsert":
      return "Ins";
    case "KeyDelete":
      return "Del";
    case "KeyPageUp":
      return "PgUp";
    case "KeyPageDown":
      return "PgDn";
    case "KeyPlus":
      return "+";
    case "KeyAsterisk":
      return "*";
    case "KeyUp":
      return "↑";
    case "KeyDown":
      return "↓";
    case "KeyLeft":
      return "←";
    case "KeyRight":
      return "→";
    default:
      return key.replace("KeyKp", "").replace("Key", "");
  }
};

export const getInputLabel = (input: InputEvent, layout: string) => {
  const fixName = (name: string) => {
    return name.replace("Pad", " Pad");
  };

  if (!input) {
    return t`Unset`;
  }

  if (input.keyboard) {
    return getKeyLabel(input.keyboard);
  }
  if (input.mouse) {
    if (input.mouse.motion) {
      return t`Mouse Move`;
    }
    switch (input.mouse.button) {
      case "Left":
        return "Left Click";
      case "Right":
        return "Right Click";
      case "Middle":
        return "Middle Click";
      case "Extra1":
        return "Mouse Button 4";
      case "Extra 2":
        return "Mouse button 5";
      default:
        return input.mouse.button;
    }
  }
  if (input.touchpad) {
    if (input.touchpad.touch.button) {
      switch (input.touchpad.touch.button) {
        case "Press":
          return fixName(input.touchpad.name) + " Press";
        case "Touch":
          return fixName(input.touchpad.name) + " Touch";
      }
    }
    if (input.touchpad.touch.motion) {
      return fixName(input.touchpad.name) + " Move";
    }
  }
  if (input.gamepad) {
    if (input.gamepad.axis) {
      switch (input.gamepad.axis.name) {
        case "LeftStick":
          return "Left Stick";
        case "RightStick":
          return "Right Stick";
        default:
          return input.gamepad.axis.name;
      }
    }
    if (input.gamepad.trigger) {
      switch (input.gamepad.trigger.name) {
        case "LeftTrigger":
          return "Left Trigger";
        case "RightTrigger":
          return "Right Trigger";
        default:
          return input.gamepad.trigger.name;
      }
    }
    if (!input.gamepad.button) {
      return "?";
    }
    switch (input.gamepad.button) {
      case "LeftStick":
        return "L3 Button";
      case "RightStick":
        return "R3 Button";
      case "LeftBumper":
        return "Left Bumper";
      case "RightBumper":
        return "Right Bumper";
      default:
        return convertOrientationToButton(input.gamepad.button, layout);
    }
  }
  return "-";
};

/**
 * Given a button position, return the associated button label for a given controller.
 */
export const convertOrientationToButton = (label: string, layout?: string) => {
  if (layout === "ps5") {
    switch (label) {
      case "Start":
        return "Options";
      case "Select":
        return "Create";
      case "North":
        return "Square";
      case "West":
        return "Triangle";
      case "East":
        return "Circle";
      case "South":
        return "Cross";
      default:
        return label;
    }
  }
  if (layout === "xbox") {
    switch (label) {
      case "North":
        return "X";
      case "West":
        return "Y";
      case "East":
        return "B";
      case "South":
        return "A";
      default:
        return label;
    }
  }
  return label;
};

export const mappingCmp = (mapping1?: InputEvent, mapping2?: InputEvent) => {
  if (!mapping1 || !mapping2) {
    return false;
  }

  if (mapping1.keyboard && mapping2.keyboard) {
    return mapping1.keyboard === mapping2.keyboard;
  }
  if (mapping1.mouse && mapping2.mouse) {
    if ("button" in mapping1.mouse && "button" in mapping2.mouse) {
      return mapping1.mouse.button === mapping2.mouse.button;
    }
    if ("motion" in mapping1.mouse && "motion" in mapping2.mouse) {
      return true;
    }
  }
  if (mapping1.touchpad && mapping2.touchpad) {
    if (mapping1.touchpad.name !== mapping2.touchpad.name) {
      return false;
    }
    if (
      "button" in mapping1.touchpad.touch &&
      "button" in mapping2.touchpad.touch
    ) {
      return (
        mapping1.touchpad.touch.button === mapping2.touchpad.touch.button &&
        mapping1.touchpad.name === mapping2.touchpad.name
      );
    }
    if (
      "motion" in mapping1.touchpad.touch &&
      "motion" in mapping2.touchpad.touch
    ) {
      return true;
    }
  }
  if (mapping1.gamepad && mapping2.gamepad) {
    if ("button" in mapping1.gamepad && "button" in mapping2.gamepad) {
      return mapping1.gamepad.button === mapping2.gamepad.button;
    }
    if (mapping1.gamepad.trigger && mapping2.gamepad.trigger) {
      return mapping1.gamepad.trigger.name === mapping2.gamepad.trigger.name;
    }
    if (mapping1.gamepad.axis && mapping2.gamepad.axis) {
      if (mapping1.gamepad.axis.direction !== mapping2.gamepad.axis.direction) {
        return false;
      }
      return mapping1.gamepad.axis.name === mapping2.gamepad.axis.name;
    }
  }
  return false;
};

export const getPhysicalLayoutFromDevice = (device: ControllerInfo) => {
  // Xbox360
  if (device.capabilities.length === 20) {
    return physicalLayouts.Xbox;
  }
  if (device.capabilities.length === 21) {
    return physicalLayouts.Xbox;
  }
  // Xbox One
  if (device.capabilities.length === 25) {
    return physicalLayouts.Xbox;
  }
  if (device.capabilities.length === 33) {
    return physicalLayouts.PS5;
  }
  // Aya Neo 2
  // if (device.capabilities.length === 155) {
  //   return physicalLayouts.SteamDeck;
  // }
  if (device.capabilities.length === 173) {
    return physicalLayouts.SteamDeck;
  }
  if (device.capabilities.length === 186) {
    return physicalLayouts.ROGAlly;
  }
  console.log("Unknown device", device);
  console.log("Device capabilities", device.capabilities);
  return physicalLayouts.Generic;
};
