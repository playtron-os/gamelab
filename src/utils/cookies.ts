export const DEFAULT_COOKIE_DURATION = 1;

export const setCookie = (
  cname: string,
  cvalue: string,
  exdays: number = DEFAULT_COOKIE_DURATION
) => {
  const d = new Date();
  d.setTime(d.getTime() + exdays * 24 * 60 * 60 * 1000);
  const expires = `expires=${d.toUTCString()}`;
  const domain = getDomain();
  const cookieString = `${cname}=${cvalue};${expires};path=/${domain}`;
  document.cookie = cookieString;
};

export const getCookie = (cname: string) => {
  const name = `${cname}=`;
  const ca = document.cookie.split(";");
  for (let i = 0; i < ca.length; i++) {
    let c = ca[i];
    while (c.charAt(0) === " ") {
      c = c.substring(1);
    }
    if (c.indexOf(name) === 0) {
      return c.substring(name.length, c.length);
    }
  }
  return "";
};

export const deleteCookie = (cname: string) => {
  const domain = getDomain();
  document.cookie = `${cname}=; expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/${domain};`;
};

function getDomain() {
  return window.location.host.includes("localhost")
    ? ""
    : `;domain=${window.location.host.split(".").slice(-2).join(".").split(":")[0]}`;
}
