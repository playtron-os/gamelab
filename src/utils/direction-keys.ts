export const getStickDirectionKey = (stickKey: string, direction: string) => {
  return `${stickKey}-${direction}`;
};
