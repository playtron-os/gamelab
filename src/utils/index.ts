export * from "./auth";
export * from "./test-utils";
