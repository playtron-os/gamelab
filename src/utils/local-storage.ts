import { error } from "@tauri-apps/plugin-log";

const ONE_WEEK_MILLISECONDS = 1000 * 60 * 60 * 24 * 7;

type LocalStorageKey = "user_id" | "last_ip"; // | 'otherKey' | 'anotherKey';

export const setInLocalStorage = <T>(key: LocalStorageKey, value: T) => {
  if (typeof window !== "undefined") {
    const item = {
      value,
      expiresIn: Date.now() + ONE_WEEK_MILLISECONDS
    };
    localStorage.setItem(key, JSON.stringify(item));
  }
};

export const parseJson = (json: string) => {
  try {
    return JSON.parse(json);
  } catch (e) {
    console.log("error parsing json", { e });
    error("failed to parse local storage json value");
    return null;
  }
};

export const getFromLocalStorage = (key: LocalStorageKey, remove = true) => {
  if (typeof window !== "undefined") {
    const item = localStorage.getItem(key);
    if (item) {
      const parsedItem = parseJson(item);

      if (Date.now() < parsedItem?.expiresIn) {
        return parsedItem.value;
      }

      if (remove) {
        localStorage.removeItem(key);
      }
    }
  }
  return null;
};

export const removeFromLocalStorage = (key: LocalStorageKey) => {
  if (typeof window !== "undefined") {
    localStorage.removeItem(key);
  }
};
