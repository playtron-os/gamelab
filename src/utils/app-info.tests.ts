import { nearestPowerOfTwo } from "./app-info";

describe("Disk size", () => {
  test("Nearest power of two", () => {
    expect(nearestPowerOfTwo(510)).toBe(512);
    expect(nearestPowerOfTwo(513)).toBe(512);
    expect(nearestPowerOfTwo(63)).toBe(64);
    expect(nearestPowerOfTwo(64)).toBe(64);
  });
});
