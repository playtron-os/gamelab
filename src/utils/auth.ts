import { getFromLocalStorage, setInLocalStorage } from "./local-storage";

export const setUserIdToLocalStorage = (userId?: string) => {
  return setInLocalStorage("user_id", userId);
};

export const getUserIdFromLocalStorage = () => {
  return getFromLocalStorage("user_id");
};
