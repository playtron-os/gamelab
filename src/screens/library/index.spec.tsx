import React from "react";
import { LibraryScreen } from ".";
import { renderWithAppLibraryContext } from "@/utils";

it("screen - renders correctly", () => {
  const { asFragment } = renderWithAppLibraryContext(<LibraryScreen />);

  expect(asFragment).toMatchSnapshot();
});
