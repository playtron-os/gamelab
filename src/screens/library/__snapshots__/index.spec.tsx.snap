// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen - renders correctly 1`] = `[Function]`;
